app/controllers/api/v1/targets_controller.rb

```diff
@@ -16,6 +16,10 @@ def destroy
         targets.find(params[:id]).destroy!
       end
 
+      def destroy
+        user.targets.find(params[:id]).destroy!
```

Comment:
You've already declared the helper `targets`, you can use it here.
I think it was added in the PR of delete targets but it seems it was overridden here.